### PR TITLE
various small fixes + 3D extension of deblurring/downsampling

### LINKED ISSUE #1167: Inpainting with non-binary masks
### Environment

all

### Describe the bug

The Inpainting class generates wrong measurements when the masks are not binary, due to the noise masking [here](https://github.com/deepinv/deepinv/blob/1d4d6f83303632893704ff5b16aac29050585cf2/deepinv/physics/inpainting.py#L133) which should be applied only if the mask is *exacty* zero.

### Steps to reproduce

```
import deepinv as dinv
import torch

device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")

img_size = (1, 3, 32, 32)

mask = torch.ones(img_size, device=device)
mask[..., :16] = 0.001

physics = dinv.physics.Inpainting(mask=mask, device=device, img_size=img_size[1:])

x = torch.ones(img_size, device=device)

y = physics(x)

x_dagger = physics.A_dagger(y)

assert x_dagger==x
```

(should be true, is false)

### Expected behavior

The mask should be applied to the noise only if it is exactly zero.

### Full traceback (if applicable)

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tachella**
### Mini-fixes
- download pretrained weights for `deepinv.models.GSDRUNet` by default, just like all other denoisers 
- remove mentions to `pyiqa` and fix range check 
- set `deepinv.optim.L2` data fidelity to `deepinv.sampling.DiffDPIR` by default
- raise error when using `deepinv.sampling.DPS` with > 1000 iterations
- Fixes inpainting bug with non-binary masks #1167 
- removes unused variables in `minres` solver. Adds missing mentions to `minres` in docstrings
 
###  Extends blur/downsampling operators
- extends definition of `sinc`, `gaussian_blur`, `bilinear`, `bicubic` filters to n-dimensions.

### Checks to be done before submitting your PR

- [ ] `python3 -m pytest deepinv/tests` runs successfully.
- [ ] `black .` and `ruff check .` run successfully.
- [ ] `make html` runs successfully (in the `docs/` directory).
- [ ] Updated docstrings related to the changes (as applicable).
- [ ] Added an entry to the [changelog.rst](https://github.com/deepinv/deepinv/blob/main/docs/source/changelog.rst).

### CODE REVIEW

File: deepinv/physics/blur.py [hunk 1]
```diff
@@ -292,6 +331,10 @@ def A_adjoint(
         r"""
         Adjoint operator of the downsampling operator.
 
+        Supports n-dimensional inputs: ``y`` can be 4D ``(B, C, H', W')`` for 2D or
```
Comment (Andrewwango) [inline deepinv/physics/blur.py:334]: ` instead of ``

File: deepinv/physics/blur.py [hunk 2]
```diff
@@ -292,6 +331,10 @@ def A_adjoint(
...
+        5D ``(B, C, D', H', W')`` for 3D, etc., where the primed dimensions are the
+        downsampled spatial dimensions.
```
Comment (Andrewwango) [inline deepinv/physics/blur.py:336]:
```suggestion
        5D ``(B, C, D', H', W')`` for 3D, etc.
```

File: deepinv/physics/blur.py [hunk 3]
```diff
@@ -302,43 +345,61 @@ def A_adjoint(
             If `factor` is passed, `filter` must also be passed as a `str` or `Tensor`, in order to update the filter to the new factor.
 
         """
-        self.imsize_dynamic = (
-            y.shape[-3],
-            y.shape[-2] * self.factor,
-            y.shape[-1] * self.factor,
+        ndim_spatial = y.dim() - 2  # number of spatial dimensions
```
Comment (Andrewwango) [inline deepinv/physics/blur.py:348]:
```suggestion
        ndim_spatial = y.dim() - 2
```

File: deepinv/physics/blur.py [hunk 4]
```diff
@@ -302,43 +345,61 @@ def A_adjoint(
...
+
+        # imsize_dynamic = (C, *original_spatial_dims) — upsample each spatial dim by factor
```
Comment (Andrewwango) [inline deepinv/physics/blur.py:350]:
```suggestion
```

File: deepinv/physics/blur.py [hunk 5]
```diff
@@ -302,43 +345,61 @@ def A_adjoint(
...
 
         imsize = self.imsize if self.imsize is not None else self.imsize_dynamic
 
         if self.filter is not None and self.padding == "valid":
-            imsize = (
-                imsize[0],
-                imsize[1] - self.filter.shape[-2] + 1,
-                imsize[2] - self.filter.shape[-1] + 1,
+            # For valid padding the blurred (pre-downsampled) spatial size is smaller
+            # than the original: each spatial dim i shrinks by (filter_size[i] - 1).
```
Comment (Andrewwango) [inline deepinv/physics/blur.py:360]:
```suggestion
```

File: deepinv/physics/blur.py [hunk 6]
```diff
@@ -302,43 +345,61 @@ def A_adjoint(
...
+            spatial_adj = tuple(
```
Comment (Andrewwango) [inline deepinv/physics/blur.py:361]: unnecessary auxiliary var

File: deepinv/physics/blur.py [hunk 7]
```diff
@@ -302,43 +345,61 @@ def A_adjoint(
...
+                for i in range(ndim_spatial)
             )
-        else:
-            imsize = imsize[:3]
+            imsize = (imsize[0],) + spatial_adj
 
         x = torch.zeros((y.shape[0],) + imsize, device=y.device, dtype=y.dtype)
-        x[:, :, :: self.factor, :: self.factor] = y  # upsample
+
+        # n-D upsampling: insert y into the stride-factor positions
```
Comment (Andrewwango) [inline deepinv/physics/blur.py:369]: please remove unnecessary comments and unnecessary auxibiliary variables to aid readability

File: deepinv/physics/blur.py [hunk 8]
```diff
@@ -302,43 +345,61 @@ def A_adjoint(
...
-            )  # Note: this may be slow against x = conv_transpose2d_fft(x, self.filter) in the case of circular padding
+            if ndim_spatial == 2:
+                x = dF.conv_transpose2d(
+                    x, self.filter, padding=self.padding
+                )  # Note: this may be slow against x = conv_transpose2d_fft(x, self.filter) in the case of circular padding
+            elif ndim_spatial == 3:
+                x = dF.conv_transpose3d(x, self.filter, padding=self.padding)
+            else:
+                raise ValueError(
+                    f"Unsupported spatial dimensionality {ndim_spatial}. "
```
Comment (Andrewwango) [inline deepinv/physics/blur.py:384]: this raise should happen earlier?/

File: deepinv/physics/blur.py [hunk 9]
```diff
@@ -302,43 +345,61 @@ def A_adjoint(
...
+        If the padding is circular and input is a 2D image, i.e., of size `(B, C, H, W)`,
+        it computes the proximal operator with the closed-formula of :footcite:t:`zhu2014fast`.
 
         Otherwise, it computes it using the conjugate gradient algorithm which can be slow if applied many times.
         """
 
-        if use_fft and self.padding == "circular":  # Formula from (Zhao, 2016)
+        if (
+            y.ndim == 4 and use_fft and self.padding == "circular"
+        ):  # Formula from (Zhao, 2016)
```
Comment (Andrewwango) [inline deepinv/physics/blur.py:402]: change citation to in docstring

**Review by Andrewwango — CHANGES_REQUESTED** [review]
general comments
- please remove unnecessary comments that just describe the code and provide no other info
- please remove unnecessary auxiliary variables that are used only once and hurt readability
- please switch `` to ` as per our convention
